lib/engine/tile.rb

```diff
@@ -360,6 +360,43 @@ def compute_city_town_edges
       ct_edges
     end
 
+    # this is invariant over rotations
+    def crossover?
+      return @_crossover if defined?(@_crossover)
+
+      @_crossover = compute_crossover
+    end
+
+    def compute_crossover
+      edge_paths = Hash.new { |h, k| h[k] = [] }
+      @paths.each do |p|
+        edge_paths[p.a.num].append(p) if p.a.edge?
+        edge_paths[p.b.num].append(p) if p.b.edge?
+        next unless p.nodes.size == 1
+
+        ct_edge = preferred_city_town_edges[p.nodes.first]
```

Comment:
what's the point of these 3 lines? they don't do anything

Reply:
It calls those methods in path with the proper city/town thus causing them to cache. I don't want to call preferred_city_town_edges in Path.

Reply:
what's the point of caching them here? why don't you just do it lazily?

Reply:
ohhhhh, wait what, you rely on the order for caching? that's a really dangerous pattern. the cache should take into consideration arguments

Reply:
Revamped this to not use arguments at all and to avoid this entire problem.